unity/EditorPlugin/PluginEntryPoint.cs

```diff
@@ -145,75 +145,74 @@ private static void Init(Lifetime lifetime)
-        foreach (var solutionFile in solutionFiles)
+      var solutionFiles = currentDirectory.GetFiles("*.sln", SearchOption.TopDirectoryOnly);
+      foreach (var solutionFile in solutionFiles)
+      {
+        var solutionName = Path.GetFileNameWithoutExtension(solutionFile.FullName);
+        if (!solutionName.Equals(currentDirectory.Name))
         {
-            var solutionName = Path.GetFileNameWithoutExtension(solutionFile.FullName);
-            if (!solutionName.Equals(currentDirectory.Name))
-            {
-                solutionNames.Add(solutionName);
-            }
+          solutionNames.Add(solutionName);
         }
+      }
 
-        var protocols = new List<ProtocolInstance>();
+      var protocols = new List<ProtocolInstance>();
 
-        // if any protocol connection is lost, we will drop all connections and recreate them
-        var allProtocolsLifetimeDefinition = lifetime.CreateNested();
-        foreach (var solutionName in solutionNames)
-        {
-            var port = CreateProtocolForSolution(allProtocolsLifetimeDefinition.Lifetime, solutionName, () =>
-            {
-                allProtocolsLifetimeDefinition.Terminate();
-            });
+      // If any protocol connection is lost, we will drop all connections and recreate them
+      var allProtocolsLifetimeDefinition = lifetime.CreateNested();
+      foreach (var solutionName in solutionNames)
+      {
+        var port = CreateProtocolForSolution(allProtocolsLifetimeDefinition.Lifetime, solutionName,
+          () => allProtocolsLifetimeDefinition.Terminate());
 
-            if (port == -1)
-                continue;
+        if (port == -1)
+          continue;
 
-            protocols.Add(new ProtocolInstance(solutionName, port));
-        }
+        protocols.Add(new ProtocolInstance(solutionName, port));
+      }
 
-        allProtocolsLifetimeDefinition.Lifetime.OnTermination(() =>
+      allProtocolsLifetimeDefinition.Lifetime.OnTermination(() =>
+      {
+        if (lifetime.IsAlive)
         {
-            if (lifetime.IsAlive)
-            {
-                ourLogger.Verbose("Recreating protocol, project lifetime is alive");
-                InitializeProtocol(lifetime, protocolInstancePath);
-            }
-            else
-            {
-                ourLogger.Verbose("Protocol will be recreating on next domain load, project lifetime is not alive");
-            }
-        });
+          ourLogger.Verbose("Recreating protocol, project lifetime is alive");
+          InitializeProtocol(lifetime);
+        }
+        else
+        {
+          ourLogger.Verbose("Protocol will be recreated on next domain load, plugin lifetime is not alive");
+        }
+      });
 
-        var result = ProtocolInstance.ToJson(protocols);
-        File.WriteAllText(protocolInstancePath, result);
+      ourLogger.Verbose("Writing Library/ProtocolInstance.json");
```

Comment:
let's move to first line of method or add new log: "start protocol initialization"

Reply:
Previously if somehting failed during scanning FS, we have last entry in log about protocol initialization

Reply:
Done